examples/distro/custom/src/main/java/io/opentelemetry/demo/instrumentation/DemoServlet3Instrumentation.java

```diff
@@ -0,0 +1,80 @@
+package io.opentelemetry.demo.instrumentation;
+
+import static io.opentelemetry.javaagent.tooling.ClassLoaderMatcher.hasClassesNamed;
+import static io.opentelemetry.javaagent.tooling.bytebuddy.matcher.AgentElementMatchers.safeHasSuperType;
+import static io.opentelemetry.javaagent.tooling.matcher.NameMatchers.namedOneOf;
+import static java.util.Collections.singletonMap;
+import static net.bytebuddy.matcher.ElementMatchers.isPublic;
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static net.bytebuddy.matcher.ElementMatchers.takesArgument;
+
+import com.google.auto.service.AutoService;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.javaagent.tooling.Instrumenter;
+import java.util.Map;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletResponse;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+
+/**
+ * This is a demo instrumentation which hooks into servlet invocation and modifies the http response.
+ */
+@AutoService(Instrumenter.class)
+public final class DemoServlet3Instrumentation extends Instrumenter.Default {
+  public DemoServlet3Instrumentation() {
+    super("servlet-demo", "servlet-3");
+  }
+
+  /*
+  We want this instrumentation to be applied after the standard servlet instrumentation.
+  The latter creates a server span around http request.
+  This instrumentation needs access to that server span.
+   */
+  @Override
+  public int getOrder() {
+    return 1;
+  }
+
+  @Override
+  public ElementMatcher<ClassLoader> classLoaderMatcher() {
+    return hasClassesNamed("javax.servlet.http.HttpServlet");
+  }
+
+  @Override
+  public ElementMatcher<TypeDescription> typeMatcher() {
+    return safeHasSuperType(
+        namedOneOf("javax.servlet.FilterChain", "javax.servlet.http.HttpServlet"));
+  }
+
+  @Override
+  public Map<? extends ElementMatcher<? super MethodDescription>, String> transformers() {
+    return singletonMap(
+        namedOneOf("doFilter", "service")
+            .and(takesArgument(0, named("javax.servlet.ServletRequest")))
+            .and(takesArgument(1, named("javax.servlet.ServletResponse")))
+            .and(isPublic()),
+        DemoServlet3Instrumentation.class.getName() + "$DemoServlet3Advice");
+  }
+
+  @SuppressWarnings("unused")
+  public static class DemoServlet3Advice {
+
+    public static final String X_SERVER_ID = "X-server-id";
```

Comment:
Muzzle is not configured in this example. Since there are compile-time dependencies and can it can result in runtime errors I would document this in a readme and add links to issues.

Reply:
I don't understand the concern, tbh :) Do you mean static field in Advice?

Reply:
Oh good point about no muzzle here.

I think the only reason this example works is because javac is inlining the string constant.

If doesn't inline the string constant, this should fail, because the advice class itself lives in the AgentClassLoader, while the bytecode inside of the advice methods is injected into the instrumented class, which lives in the user class loader, and so the bytecode inside of the advice methods should not be allowed to reference a static field in the advice class, and you should get a runtime error.

I'd suggest inlining the string here, otherwise it may give the impression that this is a valid pattern (e.g. for Loggers, Tracers, other static fields on Advice).

Reply:
Done